x-pack/plugins/ml/server/routes/annotations.ts

```diff
@@ -102,7 +102,10 @@ export function annotationRoutes(
         }
 
         const { indexAnnotation } = annotationServiceProvider(context);
-        const user = securityPlugin.authc.getCurrentUser(request) || {};
+
+        const currentUser =
+          securityPlugin !== undefined && securityPlugin.authc.getCurrentUser(request);
+        const user = currentUser ? currentUser : {};
```

Comment:
looking at the `User` type, `username` isn't optional, so this check could probably be 
`const username = currentUser?.username || ANNOTATION_USER_UNKNOWN;`

Reply:
Prefer the `??` operator, even if it does the same here. 
`const username = currentUser?.username ?? ANNOTATION_USER_UNKNOWN;`

Reply:
Good point - updated 6fdc3375d75d9381b53e972c7a484d13b3dfc514